debezium-core/src/main/java/io/debezium/time/NanoDuration.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.time;
+
+import java.math.BigDecimal;
+import java.time.temporal.ChronoUnit;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+/**
+ * A utility representing a duration into a corresponding {@link SchemaBuilder#float64() FLOAT64}
+ * number of <em>nanosecond</em>, and for defining a Kafka Connect {@link Schema} for duration values.
+ *
+ * The duration is amount is stored in floating point as opposed to an integer because certain DBs (e.g. Postgres) will store
+ * this amount internally into more than 8 bytes.
+ **/
+public class NanoDuration {
+
+    /**
+     * The approximation used by the plugins when converting a duration to nanos
+     */
+    public static final double DAYS_PER_MONTH_AVG = 365.25 / 12.0d;
+
+    public static final String SCHEMA_NAME = "io.debezium.time.NanoDuration";
+
+    /**
+     * Returns a {@link SchemaBuilder} for a {@link NanoDuration}. The resulting schema will describe a field
+     * with the {@value #SCHEMA_NAME} as the {@link Schema#name() name} and {@link SchemaBuilder#float64()} FLOAT64} for the literal
+     * type storing the number of <em>nanoseconds</em> for that duration.
+     * <p>
+     * You can use the resulting SchemaBuilder to set or override additional schema settings such as required/optional, default
+     * value, and documentation.
+     *
+     * @return the schema builder
+     */
+    public static SchemaBuilder builder() {
+        return SchemaBuilder.float64()
```

Comment:
@bingqinzhou What I'm wondering: can that ever be a floating point number actually, or would an integer type (int64) not suffice? It's likely copied from the existing `MicroDuration`,  but in fact I'm having doubts about that one, too. WDYT?